### PR TITLE
Feat(BCS-6): signup page backend

### PR DESCRIPTION
**Pull request by karzunn**
Add new user when user signs up, handle errors, and return to the login screen when successful

### CODE REVIEW

File: src/pages/authPage/AuthPage.tsx [hunk 1]
```diff
@@ -5,10 +5,10 @@ import LogInForm from "./SignInForm";
 import unity from "../../assets/unity.svg";
 
 const AuthPage = () => {
-  const [value, setValue] = useState(0);
+  const [page, setPage] = useState(0);
```
Comment (carcamog) [inline src/pages/authPage/AuthPage.tsx]: Maybe we should change this to tab instead of page @karzunn

File: src/pages/authPage/SignUpForm.tsx [hunk 2]
```diff
@@ -18,20 +26,26 @@ const SignUpForm = () => {
 
   const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
```
Comment (carcamog) [inline src/pages/authPage/SignUpForm.tsx]: can this handleChange be more specific, handleInputChange?

**Review by carcamog — CHANGES_REQUESTED** [review]

File: src/services/user.tsx [hunk 3]
```diff
@@ -13,12 +22,36 @@ class UserService {
     constructor(auth: Auth, firestore: Firestore) {
         this.auth = auth
         this.firestore = firestore
-        this.userCollection = collection(firestore, userCollection)
+        this.userCollection = collection(firestore, userCollection.name)
+    }
+
+    async getUserByUserName(userName: string): Promise<User | null> {
+        const q = query(this.userCollection, where(userCollection.fields.userName, "==", userName));
```
Comment (carcamog) [inline src/services/user.tsx]: q?

Comment (karzunn) [reply src/services/user.tsx]: :)

**Review by carcamog — APPROVED** [review]
Looks good!

**Pull request merged**